### PR TITLE
Remove es_version crate and use SeqTypes::Base version

### PR DESCRIPTION
**Pull request by imabdulbasit**
We have consistencies where we use the es_version crate's SequencerVersion in most places, and also SeqTypes::Base version. This did not cause any issues as both versions were the same. However, updating the base version would cause some tests to fail.

This PR
- removes es_crate completely and use SeqTypes::Base version

future work:
- parameterize SeqTypes<Base, Upgrade> and then instantiate the sequencer at the top level with the appropriate Base and Upgrade version based on runtime configuration (e.g. genesis file) so that we are still able to run staging tests with 0.1 and 0.2 with the same binary
- add message compat tests for all versions as changing base version will break them

### CODE REVIEW

File: builder/src/bin/permissionless-builder.rs [hunk 1]
```diff
@@ -123,7 +122,7 @@ async fn main() -> anyhow::Result<()> {
         genesis.chain_config,
         l1_params,
         opt.state_peers,
-        <SeqTypes as NodeType>::Base::instance(),
+        StaticVersion::<0, 1>::instance(),
```
Comment (jbearer) [inline builder/src/bin/permissionless-builder.rs:125]: Why change from using `SeqTypes` to hard-coding?

File: builder/src/lib.rs [hunk 2]
```diff
@@ -86,7 +86,7 @@ pub fn run_builder_api_service(url: Url, source: ProxyGlobalState<SeqTypes>) {
     let private_mempool_api = hotshot_builder_api::v0_1::builder::submit_api::<
         ProxyGlobalState<SeqTypes>,
         SeqTypes,
-        <SeqTypes as NodeType>::Base,
+        StaticVersion<0, 1>,
```
Comment (jbearer) [inline builder/src/lib.rs:89]: Why hard code?

File: builder/src/lib.rs [hunk 3]
```diff
@@ -98,7 +98,7 @@ pub fn run_builder_api_service(url: Url, source: ProxyGlobalState<SeqTypes>) {
     app.register_module("txn_submit", private_mempool_api)
         .expect("Failed to register the private mempool API");
 
-    async_spawn(app.serve(url, <SeqTypes as NodeType>::Base::instance()));
+    async_spawn(app.serve(url, StaticVersion::<0, 1>::instance()));
```
Comment (jbearer) [inline builder/src/lib.rs:101]: Why hard code?

File: builder/src/bin/permissionless-builder.rs (hunk 1 above)
Comment (imabdulbasit) [reply builder/src/bin/permissionless-builder.rs:125]: I think  builder uses v1 api so I added V1 version. I can change it to Base and see if tests passes

Comment (imabdulbasit) [reply builder/src/bin/permissionless-builder.rs:125]: Actually, now I remember I was getting version mismatch errors and had to change to v1

Comment (jbearer) [reply builder/src/bin/permissionless-builder.rs:125]: Oh builder API version is unrelated to sequencer protocol...fair enough

**Review by jbearer — APPROVED** [review]

**Pull request merged**